template/Makefile.j2

```diff
@@ -150,9 +150,15 @@ build: regenerate-charts regenerate-nix helm-package docker-build
 
 publish: docker-publish helm-publish
 
-run-dev:
+check-nix:
+	@which nix || (echo "Error: 'nix' is not installed. Please install it to proceed."; exit 1)
+
+check-kubernetes:
+	@kubectl cluster-info > /dev/null 2>&1 || (echo "Error: Kubernetes is not running or kubectl is not properly configured."; exit 1)
+
+run-dev: check-nix check-kubernetes
 	kubectl apply -f deploy/stackable-operators-ns.yaml
 	nix run -f. tilt -- up --port {[5430 + operator_index}] --namespace stackable-operators
 
-stop-dev:
+stop-dev: check-nix
```

Comment:
As this is also accessing k8s I would argue it would make sense to check it here as well

Reply:
done! @sbernauer approve again?